apps/examples/testcase/ta_tc/preference/utc/utc_preference_main.c

```diff
@@ -269,6 +311,45 @@ static void utc_preference_get_string_n(void)
 	TC_SUCCESS_RESULT();
 }
 
+static void utc_preference_get_binary_p(void)
+{
+        int ret;
+	void *value;
+
+	BINARY_VALUE.val1 = 123;
+	BINARY_VALUE.val2 = 'x';
+
+	ret = preference_get_binary(BINARY_KEY, &value);
+        TC_ASSERT_NEQ("preference_get_binary", value, NULL);
+        TC_ASSERT_EQ_CLEANUP("preference_get_binary", ret, OK, free(value));
+        TC_ASSERT_EQ_CLEANUP("preference_get_binary", check_binary_value((struct bin_pref_val_s *)value, BINARY_VALUE), true, free(value));
+
+        /* Clean binary value allocated from preference */
+        free(value);
+
+        TC_SUCCESS_RESULT();
+}
+
+static void utc_preference_get_binary_n(void)
+{
+	int ret;
+	void *value = NULL;
+
+	ret = preference_get_binary(NULL, &value);
+	TC_ASSERT_EQ_CLEANUP("preference_get_binary", value, NULL, free(value));
+	TC_ASSERT_EQ("preference_get_binary", ret, PREFERENCE_INVALID_PARAMETER);
+
+	ret = preference_get_binary(INT_KEY, &value);
+	TC_ASSERT_EQ_CLEANUP("preference_get_binary", value, NULL, free(value));
+	TC_ASSERT_EQ("preference_get_binary", ret, PREFERENCE_INVALID_PARAMETER);
+
+	ret = preference_get_binary(INVALID_KEY, &value);
+	TC_ASSERT_EQ_CLEANUP("preference_get_binary", value, NULL, free(value));
+	TC_ASSERT_EQ("preference_get_binary", ret, PREFERENCE_KEY_NOT_EXIST);
```

Comment:
I think it is better to check the line 348 first, and check the value as line 347 after.

Reply:
Pending